### PR TITLE
`InvalidDataError` is being returned instead of `ConstraintsNotMetError` when working with constraints

### LINKED ISSUE #1833: InvalidDataError: The provided data does not match the metadata (although it matches)
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 1.10.0
* Python version: 3.10.6
* Operating System: Windows 10 22H2

### Error Description

Error when using constraint 'ScalarRange' on a numeric columns to train a synthesizer (see the attached image):

**Error message:** 
File ~\.conda\envs\scrf\lib\site-packages\sdv\single_table\base.py:164, in BaseSynthesizer.validate(self, data)
    161 errors += self._validate(data)  # Validate rules specific to each synthesizer
    163 if errors:
--> 164     raise InvalidDataError(errors)

InvalidDataError: The provided data does not match the metadata:

Data is not valid for the 'ScalarRange' constraint:
   col_with_constraint
0                         1.000000
1                         0.936195
2                         0.936195
3                         0.936195
4                         0.936195
+2656 more

**This is how the column "col_with_constraint" looks like:**
image
image

### Steps to reproduce

<Replace this text with a description of the steps that anyone can follow to reproduce the error. If the error happens only on a specific dataset, please consider attaching some example data to the issue so that others can use it to reproduce the error.>

```
from sdv.metadata import SingleTableMetadata
from sdv.single_table import GaussianCopulaSynthesizer
from sdv.datasets.local import load_csvs


datasets = load_csvs(
    folder_name = os.path.join('path_to_data_folder_with_csv_file'),
    read_csv_parameters={
        'skipinitialspace': True,
        #'encoding': 'utf_32'
    })

df = datasets['data_file']

constraint1 = {
    'constraint_class': 'ScalarRange',
    #'table_name': 'guests', # for multi table synthesizers
    'constraint_parameters': {
        'column_name': 'col_with_constraint',
        'low_value': 0.7,
        'high_value': 0.9,
        'strict_boundaries': False
    }
}

synthesizer = GaussianCopulaSynthesizer(metadata)
synthesizer.add_constraints(constraints=[
    constraint1
])
synthesizer.fit(df)
```

### LINKED ISSUE #1842: When inappropriately applying ScalarRange constraint, InvalidDataError is being returned instead of ConstraintsNotMetError
### Environment Details

* SDV version: 0.10.0
* Python version: 3.11.x

### Error Description

If you have data outside of a range of values but you try to apply a ScalarRange constraint anyway, the `ConstraintsNotMetError` ([link](https://docs.sdv.dev/sdv/reference/support-and-troubleshooting/support-for-sdv-usage#constraintsnotmeterror)) should be thrown but instead the `InvalidDataError` is being thrown.

Originally identified here: https://github.com/sdv-dev/SDV/issues/1833

### Steps to reproduce

Quick code snippet to reproduce in sdv 0.10:

```
from sdv.datasets.demo import get_available_demos, download_demo
from sdv.single_table import CTGANSynthesizer

demos_df = get_available_demos(modality='single_table')
data, metadata_obj = download_demo('single_table', 'census_extended')

constraint = {
    'constraint_class': 'ScalarRange',
    'constraint_parameters': {
        'column_name': 'age',
        'low_value': 5,
        'high_value': 10,
        'strict_boundaries': True
    }
}

synthesizer = CTGANSynthesizer(metadata_obj, epochs=500, verbose=True)
synthesizer.add_constraints(constraints=[constraint])
synthesizer.fit(data)
```

This is the resulting error:

```
---------------------------------------------------------------------------
InvalidDataError                          Traceback (most recent call last)
[<ipython-input-4-463f9f3b4286>](https://localhost:8080/#) in <cell line: 16>()
     14 synthesizer = CTGANSynthesizer(metadata_obj, epochs=500, verbose=True)
     15 synthesizer.add_constraints(constraints=[constraint])
---> 16 synthesizer.fit(data)

2 frames
[/usr/local/lib/python3.10/dist-packages/sdv/single_table/base.py](https://localhost:8080/#) in fit(self, data)
    393         self._data_processor.reset_sampling()
    394         self._random_state_set = False
--> 395         processed_data = self._preprocess(data)
    396         self.fit_processed_data(processed_data)
    397 

[/usr/local/lib/python3.10/dist-packages/sdv/single_table/ctgan.py](https://localhost:8080/#) in _preprocess(self, data)
    211 
    212     def _preprocess(self, data):
--> 213         self.validate(data)
    214         self._data_processor.fit(data)
    215         self._print_warning(data)

[/usr/local/lib/python3.10/dist-packages/sdv/single_table/base.py](https://localhost:8080/#) in validate(self, data)
    162 
    163         if errors:
--> 164             raise InvalidDataError(errors)
    165 
    166     def _validate_transformers(self, column_name_to_transformer):

InvalidDataError: The provided data does not match the metadata:

Data is not valid for the 'ScalarRange' constraint:
   age
0   39
1   50
2   38
3   53
4   28
+32556 more
```

### LINKED ISSUE #1856: When inappropriately applying a CustomConstraint, an InvalidDataError is being returned instead of ConstraintsNotMetError
_This is potentially related to #1842_

### Environment Details
* SDV version: 1.10.0

### Error Description
A user was recently trying to add a CustomConstraint, but the constraint they provided was not valid for all the rows of the real data. When attempting to fit the data, the user shared that the following message was printed:

```
InvalidDataError: The provided data does not match the metadata:

Data is not valid for the 'CustomConstraint' constraint:
                     Column_A       Column_B
1238924                     9             19
2258924                     3             12
2358930                     1             10
```

This message let to a lot of confusion. It implies there is a mismatch between the data and metadata but the actual issue is that the constraint is not valid for all the rows.

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #1856 
Resolves #1842 
Resolves #1833 

CU-86azqywf4

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86azqywf4 SDV - When inappropriately applying a CustomConstraint, an InvalidDataError is being returned instead of ConstraintsNotMetError #1856](https://app.clickup.com/t/86azqywf4)

**Review by frances-h — APPROVED** [review]

File: sdv/multi_table/base.py [hunk 1]
```diff
@@ -224,19 +224,22 @@ def validate(self, data):
...
+
                 # Validate rules specific to each synthesizer
                 errors += self._table_synthesizers[table_name]._validate(data[table_name])
 
-        if errors:
+        if constraints_errors:
+            raise ConstraintsNotMetError(constraints_errors)
+
+        elif errors:
             raise InvalidDataError(errors)
```
Comment (rwedge) [inline sdv/multi_table/base.py:243]: Do we have a test that enters this code block?

**Review by rwedge — APPROVED** [review]
Looks good, left one comment

File: sdv/multi_table/base.py (hunk 1 above)
Comment (pvk-developer) [reply sdv/multi_table/base.py:243]: Now we do :)

**Pull request merged**